tools/src/wyvern/tools/typedAST/core/declarations/DefDeclaration.java

```diff
@@ -355,5 +427,12 @@ public static DefDeclaration generateSetter (GenContext ctx, wyvern.tools.typedA
 		return setterDecl;
 		
 	}
-	
+
+    public static StructuralType genericStructuralType() {
+        List<DeclType> bodyDecl = new LinkedList<>(); // these are the declarations internal to the struct
+        bodyDecl.add(new AbstractTypeMember(GENERIC_MEMBER)); // the body contains only a abstract type member representing the generic type
```

Comment:
I'm going to see how inference affects this first. I'm anticipating inference complicating this change.

If there's no sentinel variable name, we may not be able to infer when the type is a generic.
However, as long as the parameter name is sentinel, we should be fine.